modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/user/facet/template/UserFacetPortletDisplayTemplateHandler.java

```diff
@@ -0,0 +1,120 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.search.web.internal.user.facet.template;
+
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.portletdisplaytemplate.BasePortletDisplayTemplateHandler;
+import com.liferay.portal.kernel.template.TemplateHandler;
+import com.liferay.portal.kernel.template.TemplateVariableGroup;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.search.web.internal.facet.display.context.UserSearchFacetDisplayContext;
+import com.liferay.portal.search.web.internal.facet.display.context.UserSearchFacetTermDisplayContext;
+import com.liferay.portal.search.web.internal.user.facet.constants.UserFacetPortletKeys;
+import com.liferay.portlet.display.template.constants.PortletDisplayTemplateConstants;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Kevin Tan
+ */
+@Component(
+	configurationPid = "com.liferay.portal.search.web.internal.user.facet.configuration.UserFacetTemplateConfiguration",
+	configurationPolicy = ConfigurationPolicy.OPTIONAL, immediate = true,
+	property = "javax.portlet.name=" + UserFacetPortletKeys.USER_FACET,
+	service = TemplateHandler.class
+)
+public class UserFacetPortletDisplayTemplateHandler
+	extends BasePortletDisplayTemplateHandler {
+
+	@Override
+	public String getClassName() {
+		return UserSearchFacetTermDisplayContext.class.getName();
+	}
+
+	@Override
+	public String getName(Locale locale) {
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", locale, getClass());
+
+		String portletTitle = _portal.getPortletTitle(
+			UserFacetPortletKeys.USER_FACET, resourceBundle);
+
+		return LanguageUtil.format(locale, "x-template", portletTitle, false);
+	}
+
+	@Override
+	public String getResourceName() {
+		return UserFacetPortletKeys.USER_FACET;
+	}
+
+	@Override
+	public Map<String, TemplateVariableGroup> getTemplateVariableGroups(
+			long classPK, String language, Locale locale)
+		throws Exception {
+
+		Map<String, TemplateVariableGroup> templateVariableGroups =
+			super.getTemplateVariableGroups(classPK, language, locale);
+
+		TemplateVariableGroup templateVariableGroup =
+			templateVariableGroups.get("fields");
+
+		templateVariableGroup.empty();
+
+		templateVariableGroup.addVariable(
+			"term-frequency", Integer.class,
+			PortletDisplayTemplateConstants.ENTRY, "getFrequency()");
+		templateVariableGroup.addVariable(
+			"term-name", String.class, PortletDisplayTemplateConstants.ENTRY,
+			"getDisplayName()");
```

Comment:
I think I get a template error with adding getDisplayName() ? Let me know if that's the case

Reply:
Ah, good catch. That's not a valid method. I'll update it with the correct one.